### PR TITLE
Move Calendar Date Range to EDTR v2

### PR DESCRIPTION
**Pull request by tn3rb**
- see #2349
- uses Calendar Date Range in DateCard
- minor tweaks to MediumCalendarDate

### CODE REVIEW

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	footerText = '',
+	className = '',
+	showTime = false,
+}) => {
+	const startDateObject = startDate instanceof Date ? startDate : parseISO(startDate);
+	const endDateObject = endDate instanceof Date ? endDate : parseISO(endDate);
+	if (!isValid(startDateObject) || !isValid(endDateObject)) {
+		return null;
+	}
+	if (format(startDateObject, 'yy-MM-dd') !== format(endDateObject, 'yy-MM-dd')) {
```
Comment (nicolad) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx]: can this conditional be extracted into a variable?

File: assets/src/application/ui/calendars/dateDisplay/MediumCalendarDate/style.css [hunk 2]
```diff
@@ -4,20 +4,12 @@
 	font-weight: normal;
 	line-height: var(--ee-font-size-default);
 	height: 100%;
+	min-height: calc(var(--ee-padding-micro) * 45);
+	min-width: calc(var(--ee-padding-micro) * 34);
```
Comment (nicolad) [inline assets/src/application/ui/calendars/dateDisplay/MediumCalendarDate/style.css]: magic numbers :smiley:

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx (hunk 1 above)
Comment (tn3rb) [reply assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx]: sure... in fact... i think there's a date helper function that checks if dates match

File: assets/src/application/ui/calendars/dateDisplay/MediumCalendarDate/style.css (hunk 2 above)
Comment (tn3rb) [reply assets/src/application/ui/calendars/dateDisplay/MediumCalendarDate/style.css]: do you mean the 34 and 45 ?
yeah they are kind of arbitrarily set...
there's no corresponding source for those values other than that's what I felt looked best

File: assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ export const CalendarDateRange: React.FC<CalendarDateRange> = ({
 	if (!isValid(startDateObject) || !isValid(endDateObject)) {
 		return null;
 	}
-	if (format(startDateObject, 'yy-MM-dd') !== format(endDateObject, 'yy-MM-dd')) {
+	if (differenceInCalendarDays(startDateObject, endDateObject) !== 0) {
```
Comment (nicolad) [inline assets/src/application/ui/calendars/dateDisplay/CalendarDateRange/index.tsx]: Noooooice :+1:

File: assets/src/application/constants/date-fns-formats.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+export const DEFAULT_DATE_FORMAT = 'yyyy-MM-ee'; // 2020-12-31
```
Comment (nicolad) [inline assets/src/application/constants/date-fns-formats.ts]: Comments are pure gold :medal_sports:

**Pull request merged**